documents/service.go

```diff
@@ -28,6 +28,12 @@ type DocumentProof struct {
 	FieldProofs []*proofspb.Proof
 }
 
+// Patcher interface defines a Patch method for inner Models
+type Patcher interface {
+	// Patch merges payload data into model
+	Patch(payload UpdatePayload) (Model, error)
```

Comment:
Since we anyway call this on the model, we dont need to return the model I think